Tasks/AzureIoTEdgeV2/util.ts

```diff
@@ -80,25 +80,43 @@ export default class Util {
     }
 
     let cmds: Cmd[] = [];
-    let version = Constants.iotedgedevDefaultVersion;
-    if (tl.getVariable(Constants.iotedgedevLockVersionKey)) {
-      version = tl.getVariable(Constants.iotedgedevLockVersionKey);
-    }
-    tl.debug(`The specified iotedgedev version is: ${version}`);
+    let edgeDevVersion = Constants.iotedgedevDefaultVersion;
+    let simulatorVersion : string = null;
+    // install pre reqs
     if (tl.osType() === Constants.osTypeLinux) {
       cmds = [
         { path: `sudo`, arg: `apt-get update`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `-H pip3 install wheel --upgrade`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `pip3 install importlib-metadata==2.1.1`, execOption: Constants.execSyncSilentOption },
         { path: `sudo`, arg: `apt-get install -y python3-setuptools`, execOption: Constants.execSyncSilentOption },
-        { path: `sudo`, arg: `pip3 install ${Constants.iotedgedev}~=${version}`, execOption: Constants.execSyncSilentOption },
       ]
+
+      if (tl.getVariable(Constants.iotedgedevLockVersionKey)) {
```

Comment:
>tl.getVariable(Constants.iotedgedevLockVersionKey [](http://example.com/codeflow?start=10&length=49)

the code reads this variable multiple times; unless we expect it to change, maybe cache it in a local variable?